code/game/objects/items/stacks/money.dm

```diff
@@ -212,6 +239,40 @@
 		value = 4
 		return ..()
 
+/obj/item/stack/money/dollar100
+	name = "100 Dollar Bill"
+	desc = "Paper bank note valued at one-hundred dollars"
+	icon_state = "100dollar"
+	value = 100
+	amount = 1
+
+/obj/item/stack/money/dollar100/New()
+	if(amount < 50)
+		icon_state = "100dollar"
+	if(amount >= 50)
+		icon_state = "100dollar50"
+	if(amount >= 100)
+		icon_state = "100dollar100"
+	if(amount >= 300)
+		icon_state = "100dollar300"
+	if(amount == 500)
+		icon_state = "100dollar500"
+	update_icon()
+	return ..()
+
+/obj/item/stack/money/dollar100/update_icon()
+	if(amount < 50)
+		icon_state = "100dollar"
```

Comment:
switch(amount)
if(0 to 49)
icon_state = "ruble"
if(50 to 99)
icon_state = "ruble50"
if(100 to 249)
icon_state = "ruble100"
if(250 to INFINITY)
icon_state = "ruble500"